ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -1466,3 +1471,16 @@ func (r *RayClusterReconciler) updateClusterReason(ctx context.Context, instance
 	r.Log.Info("updateClusterReason", "Update CR Status.Reason", clusterReason)
 	return r.Status().Update(ctx, instance)
 }
+
+// sumGPUs sums the GPUs in the given resource list.
+func sumGPUs(resources map[corev1.ResourceName]resource.Quantity) resource.Quantity {
+	totalGPUs := resource.Quantity{}
+
+	for key, val := range resources {
+		if strings.HasSuffix(string(key), "gpu") && !val.IsZero() {
```

Comment:
Should we sum all or only the first GPU resource in the list? See [this logic](https://github.com/ray-project/kuberay/blob/bf3fd63a97bc0b61c1858545dee3233679754c11/ray-operator/controllers/ray/common/pod.go#L723-L724).

Reply:
Perhaps we could start with the first GPU resource only. Do you have any instances that have more than one type of GPU?

Reply:
I don't. First GPU resource SGTM.

Reply:
Actually resources are stored in a map. What would be the "first" here?

Reply:
> Actually resources are stored in a map. What would be the "first" here?

Good point. I have never seen any instances with more than one type of GPU, but it would be great if we could make the behavior more deterministic. Perhaps we could sort them and store them in a slice, and then iterate over that slice?

Reply:
If most of the time there's only one type of GPU, would it be simpler to just sum them all in the case where's there's more than one? But I'm also happy to sort the keys alphabetically and pick the first one's value too.

Reply:
> would it be simpler to just sum them all in the case where's there's more than one? 

Nice! Summing them all is a better idea.

Reply:
updated